index/generator/library/library.go

```diff
@@ -151,7 +164,11 @@ func parseDevfileRegistry(registryDirPath string, force bool) ([]schema.Schema,
 			// Index component validation
 			err := validateIndexComponent(indexComponent, schema.StackDevfileType)
 			if err != nil {
-				return nil, fmt.Errorf("%s index component is not valid: %v", devfileDir.Name(), err)
+				if _, ok := err.(*MissingArchError); !ok {
```

Comment:
This section looks fine, but was a little hard to understand at first what the if statement was doing.

To improve clarity, could we do something like the following?
```
if _, ok := err.(*MissingArchError); ok {
    fmt.Printf("the %s devfile has no architecture(s) mentioned\n", indexComponent.Name)
else {
    return nil, fmt.Errorf("%s index component is not valid: %v", devfileDir.Name(), err)
}
```

Reply:
Addressed, made it more readable. However not to the suggested snippet due to warning:
![Screen Shot 2021-07-07 at 3 34 58 PM](https://user-images.githubusercontent.com/31771087/124821633-f4060880-df3c-11eb-9ecc-7b282abb1237.png)

Reply:
Sounds good to me.